compiler/rustc_passes/src/liveness.rs

```diff
@@ -1590,6 +1594,32 @@ impl<'tcx> Liveness<'_, 'tcx> {
                             .emit();
                     },
                 )
+            } else if can_remove {
+                self.ir.tcx.struct_span_lint_hir(
+                    lint::builtin::UNUSED_VARIABLES,
+                    first_hir_id,
+                    hir_ids_and_spans.iter().map(|(_, pat_span, _)| *pat_span).collect::<Vec<_>>(),
+                    |lint| {
+                        let mut err = lint.build(&format!("unused variable: `{}`", name));
+                        err.multipart_suggestion(
+                            "try removing the field",
+                            hir_ids_and_spans
+                                .iter()
+                                .map(|(_, pat_span, _)| {
+                                    let span = self
+                                        .ir
+                                        .tcx
+                                        .sess
+                                        .source_map()
+                                        .span_extend_to_next_char(*pat_span, ',', true);
+                                    (span.with_hi(BytePos(span.hi().0 + 1)), String::new())
```

Comment:
Does this have to be `(span.with_hi(BytePos(span.hi().0 + 1))`? Doesn't `span` already include the `,`?

Reply:
Surprisingly, yes, it does. `span` excludes `,`.

Reply:
Documentation for `.span_extend_to_next_char()` implies the opposite, it doesn’t work without `+ 1` in practice.

Reply:
Interesting.
It looks like the `span_extend_to_next_char()` isn't used anywhere, so I think it would be a good opportunity to just fix it (and add a unit test for it).

> but it doesn’t work without + 1 in practice.

Is that covered by the UI test somehow or did you test it manually?

Reply:
I discovered it manually and _now_ it’s covered by UI test (since it shows the span).

Reply:
@michaelwoerister It looks like `.span_extend_to_line()` relies on this behaviour, actually.

Reply:
You're right!

OK, since I don't want to deal with different newline encodings, we can also just update the documentation to 
```
/// Extends the given `Span` to just before the next occurrence of `c`.
```
That seems just as useful 😉

Reply:
Done!